b2/exception.py

```diff
@@ -251,6 +251,19 @@ class UnusableFileName(B2SimpleError):
     pass
 
 
+class InvalidRange(B2Error):
+    def __init__(self, content_length, range_):
+        super(InvalidRange, self).__init__()
+        self.content_length = content_length
+        self.range_ = range_
+
+    def __str__(self):
+        return 'Cloud can only serve a range of 0-%d, while a range of %s was requested' % (
```

Comment:
Thinking out loud because these kinds of things are tricky.  What about this case:

* File is 15 bytes long.
* Request is for range 10-19
* Response will have content length 5 and Content-Range of 10-14
* The message here will be wrong.

The Content-Range that comes back from the cloud will look like `10-14/15`.  That's probably the best place to get the actual file size.

Reply:
actually the message was correct. I extended tests to make it clear.

Reply:
This works, because content-length in such case will be actually 15